File: SEMESTRUL 3/METODE AVANSATE DE PROGRAMARE/LABORATOR/lab 2 - home/PrinterTaskRunner.java
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class PrinterTaskRunner extends AbstractTaskRunner {
    private static final DateTimeFormatter dateTimeFormat= DateTimeFormatter.ofPattern("HH:mm");

    public PrinterTaskRunner(TaskRunner taskRunner) {
        super(taskRunner);
    }

    @Override
    public void executeOneTask() {
        super.executeOneTask();
        System.out.println("Task executed at "+ LocalDateTime.now().format(dateTimeFormat));
    }
}

File: SEMESTRUL 3/METODE AVANSATE DE PROGRAMARE/LABORATOR/lab 11 - homework/lab11/lab11/task/PrinterTaskRunner.cs
namespace lab11;

public class PrinterTaskRunner:AbstractTaskRunner
{
    public PrinterTaskRunner(TaskRunner taskRunner) : base(taskRunner)
    {
    }

    public override void ExecuteOneTask() 
    {   base.ExecuteOneTask();
        Console.WriteLine("ExecuteOneTask called at:"+DateTime.Now);
    } 
    public override void ExecuteAll()
    {   base.ExecuteAll();
        Console.WriteLine("ExecuteAll called at:"+DateTime.Now);
    }

    public override void AddTask(Task task)
    {   base.AddTask(task);
        Console.WriteLine("AddTask called at:"+DateTime.Now);
        
    }

    public override bool HasTasks()
    {   
        Console.WriteLine("HasTasks called at:"+DateTime.Now);
        return base.HasTasks();
    }
    
}